pkg/util/i18n/i18n.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package i18n
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+
+	"k8s.io/kubernetes/pkg/generated"
+
+	"github.com/chai2010/gettext-go/gettext"
+	"github.com/golang/glog"
+	"golang.org/x/text/language"
+)
+
+// see https://blog.golang.org/matchlang
+var supported = []language.Tag{
+	language.AmericanEnglish, // en-US: first language is fallback
+}
+
+var (
+	lang language.Tag
+	lock sync.Mutex
+)
+
+func loadSystemLanguage() string {
+	langStr := os.Getenv("LANG")
+	if langStr == "" {
+		glog.V(3).Infof("Couldn't find the LANG environment variable, defaulting to en-US")
+		return "default"
+	}
+	pieces := strings.Split(langStr, ".")
+	if len(pieces) == 0 {
+		glog.V(3).Infof("Unexpected system language (%s), defaulting to en-US", langStr)
+		return "default"
+	}
+	return pieces[0]
+}
+
+// LoadTranslations loads translation files. getLanguageFn should return a language
+// string (e.g. 'en-US'). If getLanguageFn is nil, then the loadSystemLanguage function
+// is used. translationFiles are paths to translation files that are expected to be
+// compiled into the binary via go-bindata.
+func LoadTranslations(root string, getLanguageFn func() string, translationFiles []string) error {
```

Comment:
In future we'll want a `getLanguageFn` that loads from preferences or something similar, so that I can manually set it.